OpenTrons/opentrons-api
# feat(protocol-designer): add export warning modal

# Overview

If a user exports a protocol that doesn't use some hardware (instruments, modules, staging areas, trash entities), we warn them on export with an itemized list of any implicated hardware. Here, I update the utility for getting content for this modal depending on the specific unused hardware. I also wire up the modal using our `useBlockingHint` hook in `ProtocolOverview`

Closes AUTH-818

## Test Plan and Hands on Testing

- create a protocol with no steps
- export and verify that "protocol has no steps" warning modal shows
- exit the modal and add steps and some hardware that is not used in the protocol (i.e., gripper without using a move labware, heater shaker without adding any heater-shaker steps, etc.).
- export again
- verify that 'protocol has unused hardware' warning modal shows
- check box to not show again and export
- start a new session and import the protocol that you just exported
- select "export" and verify that the modal does not show

## Changelog

- implement `useBlockingHint` in `ProtocolOverview` on export
- expand `HintKey` type to include new title keys for export warning modal
- add new translations 
- update tests

## Review requests

see test plan. Please verify that hint logic is correct

## Risk assessment

low
